tools/check_constitution.py

```
@@ -0,0 +1,385 @@
+        ["git", "show", f"{sha}:{path}"], cwd=ROOT, capture_output=True, text=True,
+        encoding="utf-8", errors="replace",
+    )
+    return out.stdout if out.returncode == 0 else None
+
+
+def _changed(sha: str) -> list[str]:
+    return [n for n in _git("diff", "--name-only", f"{sha}...HEAD").splitlines() if n.strip()]
+
+
+def _status_line(text: str) -> str | None:
+    return next((l for l in text.split("\n") if STATUS_LINE.match(l)), None)
```

Ruff (0.16.1)

[error] Ambiguous variable name: `l`

(E741)